agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/ActAIAgent.kt

```diff
@@ -0,0 +1,213 @@
+        feature: AIAgentNonGraphFeature<Config, Feature>,
+        configure: Config.() -> Unit
+    ) {
+        pipeline.install(feature, configure)
+    }
+    init {
+        FeatureContext(this).featureContext()
+    }
+    override suspend fun run(agentInput: Input): Output {
+        runningMutex.withLock {
+            if (isRunning) {
+                throw IllegalStateException("Agent is already running")
+            }
+            isRunning = true
+        }
+
+        val runId = Uuid.random().toString()
+
+        val llm = AIAgentLLMContext(
+            tools = toolRegistry.tools.map { it.descriptor },
+            toolRegistry = toolRegistry,
+            prompt = agentConfig.prompt,
+            model = agentConfig.model,
+            promptExecutor = PromptExecutorProxy(
+                executor = promptExecutor,
+                pipeline = pipeline,
+                runId = runId
+            ),
+            environment = environment,
+            config = agentConfig,
+            clock = clock
+        )
+
+        val context = AIAgentLoopContext(
+            environment,
+            id,
+            runId,
+            agentInput,
+            agentConfig,
+            llm,
+            AIAgentStateManager(),
+            storage = AIAgentStorage(),
+            strategyName = strategy.name,
+            pipeline = pipeline
+        )
+
+        val result = withContext(
+            AgentRunInfoContextElement(
+                agentId = id,
+                runId = runId,
+                agentConfig = agentConfig,
+                strategyName = strategy.name
+            )
+        ) {
+            strategy.execute(context, agentInput)
+        }
+
+        runningMutex.withLock {
+            isRunning = false
+        }
+
+        return result
+    }
+
+    override suspend fun close() {
+        pipeline.onAgentBeforeClosed(agentId = id)
+        pipeline.closeFeaturesStreamProviders()
+    }
+}
+
+/**
+ * Creates a new instance of `LoopAIAgent` to manage and execute AI-driven loops with a specific configuration,
+ * tool registry, and strategy for handling operations.
+ *
+ * @param promptExecutor The `PromptExecutor` responsible for processing language model prompts and managing interactions with the AI model.
+ * @param agentConfig The `AIAgentConfigBase` defining the configuration for the AI agent, including prompts, model details, and iteration limits.
+ * @param toolRegistry A `ToolRegistry` specifying the tools available to the agent. If no tools are provided, it defaults to an empty registry.
+ * @param loop A suspendable lambda function representing the custom loop execution strategy. It takes an input of type `Input` and a context of type `AIAgentLoopContext`, and returns
+ *  an output of type `Output`.
+ * @return A `LoopAIAgent` instance initialized with the specified prompt executor, configuration, tool registry, and loop strategy.
+ */
+public fun <Input, Output> actAIAgent(
```

Comment:
Let's make the creation of AIAgent uniform -- so that the switching is easy, and learning curve is smoother:

You would always write:
```
AIAgent(
        promptExecutor = executor,
        strategy = <some strategy>,
        agentConfig = agentConfig,
        toolRegistry = toolRegistry
) 
```
(or the other constructor of `AIAgent` class)


And then -- only the strategy type would determine how the agent works. 

It would also allow experimenting and switching strategies much easier

Reply:
```
AIAgent(
        promptExecutor = executor,
        strategy = loopStrategy { ... } , // or let's probably think about some better name. I don't have any good options, though
        agentConfig = agentConfig,
        toolRegistry = toolRegistry
) 
```